src/amber/controller/error.cr

```diff
@@ -27,6 +27,18 @@ module Amber::Controller
       end
     end
 
+    private def internal_server_error_message
+      begin
+        message = @ex.inspect_with_backtrace
+      rescue ex : IndexError
+        # ERROR: Any of these will cause another IndexError (at least on macOS)
+        # ex.inspect_with_backtrace # or @ex.inspect_with_backtrace
+        # ex.backtrace? # or @ex.backtrace?
+        message = @ex.message
+      end
+      return message
```

Comment:
@elorest - The message from the original exception is returned in the case that `inspect_with_backtrace` raises an `IndexError`

The first priority is still `@ex.inspect_with_backtrace` and if that doesn't work, the second priority is `@ex.message`

line 39 runs in either case.

Reply:
Will this not work?

```crystal
private def internal_server_error_message
  message = @ex.inspect_with_backtrace
rescue ex : IndexError
  # ERROR: Any of these will cause another IndexError (at least on macOS)
  # ex.inspect_with_backtrace # or @ex.inspect_with_backtrace
  # ex.backtrace? # or @ex.backtrace?
  @ex.message
end
```

Also it might be worth to use the flags just like in the crystal source code `flag?(:freebsd) || flag?(:linux) || flag?(:openbsd)` instead of rescueing

Reply:
Yeah, this should work:

```crystal
private def internal_server_error_message
  @ex.inspect_with_backtrace
rescue ex : IndexError
  @ex.message
end
```

I prefer not to count on flags in this case. It is a bug in crystal, which has been resolved in master:
https://github.com/crystal-lang/crystal/issues/5224
https://github.com/crystal-lang/crystal/commit/5eecd57ce471f3b89b4ed7ed3d54202273bc3ebc